test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriBuilder/FilterAndOrderByBuilderTests.cs

```diff
@@ -465,6 +465,14 @@ public void BuildFilterWithDollarItInFunctionCall()
             Uri actualUri = UriBuilder(queryUri, ODataUrlKeyDelimiter.Parentheses, settings);
             Assert.Equal(new Uri("http://gobbledygook/People(1)/RelatedSSNs?$filter=endswith%28%24it%2C%27xyz%27%29"), actualUri);
         }
+
+        [Fact]
+        public void BuildFilterWithManyBinaryOperator()

```

Comment:
>BuildFilterWithManyBinaryOperator [](start = 20, length = 33)

would also be good to add a test that validates the range variable logic; i.e.,
Uri queryUri = new Uri("People?$filter=MyPaintings/any(a:a/OpenProperty eq 1 or a/OpenProperty eq 2 and a/OpenProperty lt $it/ID)", UriKind.Relative)